apps/src/contrast.cpp

```diff
@@ -3,35 +3,90 @@
 #include <opencv2/highgui.hpp>
 #include <opencv2/imgproc.hpp>
 
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+
 #include "envitools/Box.hpp"
 #include "envitools/CSVIO.hpp"
 #include "envitools/ContrastMetrics.hpp"
 #include "envitools/EnviUtils.hpp"
 
 namespace fs = boost::filesystem;
 
+namespace po = boost::program_options;
+
 namespace et = envitools;
 
-// argv[1] == directory of wavelengths to be passed
-int main(int argc, char** argv)
+// argv[1] == directory of wavelengths to be passed,
+// argv[2] == output path for csv file
+// optional argv[4] == ink points file (if Michelson), ROI file (if RMS)
+// optional argv[5] == papyrus points file (if Michelson)
+int main(int argc, char* argv[])
 {
-    if (argc < 3) {
-        std::cout << " Usage: " << argv[0];
-        std::cout << " tif_directory/ output.csv" << std::endl;
+    ///// Parse the cmd line /////
+    // fs::path img_dir = argv[1];
+    // fs::path csv_path = argv[2];
+    fs::path img_dir, csv_path, inkpts_path, papypts_path, roi_path;
+
+    try {
+        // All command line options
+        // clang-format off
+        po::options_description options("Options");
+        options.add_options()
+            ("help,h","Show this message")
+            ("input-dir,i",po::value<std::string>()->required(),
+                "Input directory of wavelengths")
+            ("inkpoints,k",po::value<std::string>(),
```

Comment:
Change this to `foreground-pts,f`. Won't necessarily be used to measure ink contrast. Could just be used to measure contrast in general.